Feature/nigel

NigelAlvaro opened the pull request:
Added the following:
 - Input Variables
 - Introduce the Input Variable "Prefix"
 - Edited some other stuff to meet terraform best practice coding style

NigelAlvaro committed:
Add files via upload

SSH into EC2

NigelAlvaro committed:
Add files via upload

makotobot98 commented on main.tf:
```diff
@@ -0,0 +1,47 @@
+  name = "${var.prefix}-bastion-host" 
+  description = "security group access for all" 
+  
+ingress {
```
indent

NigelAlvaro replied:
Fixed Indent

makotobot98 commented on main.tf:
```diff
@@ -0,0 +1,47 @@
+}
+
+resource "aws_key_pair" "demokeypair" {
+  key_name   = "${var.prefix}-KeyPair"
```
naming convention for aws:
1. lowercase letters
2. use dash "-" to separate someDayWillBeSunny, some-day-will-be-sunny

NigelAlvaro replied:
Fixed naming

makotobot98 commented on variables.tf:
```diff
@@ -0,0 +1,69 @@
+    instance_type = string
+  })
+  default = {
+    ami = "ami-0925fd223898ee5ba"
```
Instead of passing a hard coded string for ami. This is what I want:
1. I want to derive ami id based on some filters:
 a. its Amazon Linux2 (or Ubuntu 22.04, or Windows)
 b. I want the latest of such AMI available in AWS
2. This can be achieved by using terraform data source https://developer.hashicorp.com/terraform/language/data-sources
3. Input variable will only require user to pass in Operating system type (Amazon Linux 2, Ubuntu 22.04, or WIndows), then in the code it should derive the corresponding AMI. This means instead of defaulting an "ami" attribute, you should default an attribute called "os_type"

NigelAlvaro replied:
You got it

makotobot98 commented on variables.tf:
```diff
@@ -0,0 +1,69 @@
+  })
+
+  default = {
+        from_port = 0
```
indent should be consistent. You should run this command "terraform fmt" it will format your terraform code

NigelAlvaro replied:
Indent fixed

NigelAlvaro committed:
#3

Added 
- Data Source for AMI
- User can now input OS_type and get the right AMI
- Indent Fixed

makotobot98 commented on variables.tf:
```diff
@@ -0,0 +1,72 @@
+  }
+}
+
+data "aws_ami" "ami_type" {
```
this file is variables.tf

makotobot98 commented on variables.tf:
```diff
@@ -0,0 +1,72 @@
+  description = "Credentials needed to connect to my AWS account"
+}
+
+locals {
```
this file is variables.tf

makotobot98 commented on variables.tf:
```diff
@@ -0,0 +1,72 @@
+locals {
+  os_type = {
+    linux = "amzn2-ami-kernel-5.10-hvm-*-x86_64-gp2"
+    ubuntu = "ubuntu/images/hvm-ssd/ubuntu-jammy-22.04-amd64-server-20230325"
```
wildcard

makotobot98 commented on variables.tf:
```diff
@@ -0,0 +1,72 @@
+}
+
+variable "os_type" {
+  type = string
```
provide descriptions on what are the available os_type for this module

makotobot98 commented:
Write a readme file that includes:
1. what does this module provision
2. how to configure credentials to aws terraform
3. how to provision (what are the necessary input variables)
4. how to connect to the bation host (how to get the private key, how to ssh)

NigelAlvaro committed:
Update Readme.md

NigelAlvaro committed:
Update #4

NigelAlvaro committed:
update variables

NigelAlvaro committed:
Update #5

NigelAlvaro committed:
update #5

NigelAlvaro committed:
update #5

NigelAlvaro committed:
update #5

NigelAlvaro committed:
Update #5
